hub/integrations/pytorch.py

```diff
@@ -57,19 +62,38 @@ def _read_and_store_chunk(
     return chunk_size
 
 
-def dataset_to_pytorch(dataset, transform: Callable = None, workers: int = 1):
+def dataset_to_pytorch(
+    dataset,
+    transform: Optional[Callable] = None,
+    workers: int = 1,
+    tuple_fields: Optional[List] = None,
+):
     dataset.flush()
-    return TorchDataset(dataset, transform, workers)
+    return TorchDataset(dataset, transform, workers, tuple_fields)
 
 
 class TorchDataset:
-    def __init__(self, dataset, transform: Callable = None, workers: int = 1):
+    def __init__(
+        self,
+        dataset,
+        transform: Optional[Callable] = None,
+        workers: int = 1,
+        tuple_fields: Optional[List[str]] = None,
+    ):
         self._import_torch()
         self.transform: Optional[Callable] = transform
         self.workers: int = workers
+        self.tuple_fields = tuple_fields
         self.map = ProcessPool(nodes=workers).map
         self.length = len(dataset)
-        self.keys = list(dataset.tensors)
+        self.keys: List[str] = list(dataset.tensors)
+        if tuple_fields is not None:
+            for field in tuple_fields:
+                if field not in self.keys:
+                    raise TensorDoesNotExistError(field)
+            unused_tensors = [k for k in self.keys if k not in tuple_fields]
+            if unused_tensors:
+                warnings.warn("Unused tensors: %s." % (", ".join(unused_tensors)))
```

Comment:
Not sure if we should have a warning here, it's good to use a subset of tensors in many cases for speedup. I remember in Hub v1, by doing a subset and removing tensors that contained meta info like filename, time etc, we were able to significantly speedup training.